Manhunt.elm

```diff
@@ -56,18 +56,58 @@ update msg model =
     case msg of
         Move direction ->
             ( { model | playerPosition = updatePlayerPosition model.playerPosition direction }
-            , Random.generate UpdateLocationData (Random.int 1 6000)
+            , Random.generate RefreshLocation (Random.int 1 6000)
             )
 
-        UpdateLocationData randomNumber ->
-            ( { model | locationsData = updateLocationData randomNumber model }
+        RefreshLocation randomNumber ->
+            ( { model | locationsData = refreshLocation randomNumber model }
             , Cmd.none
             )
 
         Perform action ->
-            case action of
-                _ ->
-                    ( model, Cmd.none )
+            ( { model | locationsData = performAction model }, Cmd.none )
+
+
+performAction : Model -> LocationsData
+performAction model =
+    let
+        position : PlayerCoordinate
+        position =
+            ( model.playerPosition.lat, model.playerPosition.lon )
+
+        locationData : LocationData
+        locationData =
+            case findSafeInDict position model.locationsData of
+                Just data ->
+                    data
+
+                Nothing ->
+                    Debug.todo "Shouldn't be possible perform action if no data"
+
+        subtract : Maybe (Maybe LocationData) -> Maybe (Maybe LocationData)
```

Comment:
Here you can probably just write a `LocationData -> LocationData` or `LocationData -> Maybe LocationData` function and use `Maybe.map` or `Maybe.andThen` to plumb it